segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -521,7 +529,11 @@ public boolean isOffline() {
                     this.durableLog.add(new MergeSegmentOperation(targetSegmentId, sourceSegmentId), timer.getRemaining()));
         }
 
-        return result.thenApply(v -> sourceMetadata.getSnapshot());
+        return result.thenApply(v -> {
+            SegmentProperties sourceProperties = sourceMetadata.getSnapshot();
+            return new MergeStreamSegmentResultImpl(this.metadata.getStreamSegmentMetadata(targetSegmentId).getLength(),
```

Comment:
`MergeSegmentOperation` also has both a `getStreamSegmentOffset` and `getLength`, which can be used instead of querying the metadata. The only case where you really do need to get the target length is when we delete the source (line 519).

(I prefer not to query the metadata too much as it's shared by all internal components of the Container, and every access requires acquiring a lock)